### DISCUSSION TITLE
Update recommended prior to iOS 18.2 general availability

### DISCUSSION THREAD

**Discussion by Kyle-Reis** [microsoftconnect/ms-intune-app-sdk-ios#495] [Announcements]
iOS 18.2 (currently in Beta) contains some new features and breaking changes that could allow an organization's data to leave the boundary of the Intune managed identity:

- A new Apple Intelligence feature that will take a screenshot of the application and extract/summarize any text found.
- The new Genmoji feature, which runs out-of-process from the host application, where MAM data protections cannot be applied
- Some breaking changes that prevent the Intune iOS MAM SDK from blocking the Writing Tools feature in certain scenarios.

The latest [20.2.1 release](https://github.com/microsoftconnect/ms-intune-app-sdk-ios/releases/tag/20.2.1) for Xcode 16 and [19.7.6 release](https://github.com/microsoftconnect/ms-intune-app-sdk-ios/releases/tag/19.7.6) for Xcode 15 addresses each of these scenarios with support for screen capture blocking, Genmoji blocking, and fixes for the Writing Tools issues.

We recommend that all application developers upgrade to the latest available supported release which corresponds to the Xcode version your application is currently compiled with, and that you push your updated application to production as soon as possible to ensure the fix for this issue is available before iOS 18.2 becomes generally available. If you have any questions, please [file an issue](https://github.com/microsoftconnect/ms-intune-app-sdk-ios/issues) in this repository or reply directly to this announcement.

**Comment by robgrame** [comment]
Could you clarify what kind of scenarios are going to be affected by the new OS versions if SDK are not up-to-date ? What are the side effects if SDK are not going to be updated before upgrading the iOS version to 18.2=

**Reply by Kyle-Reis** [reply]
If an application does not update Intune prior to iOS 18.2 GA, in general the app will continue to function the same, but certain scenarios will be unprotected by Intune. The specific scenarios are called out above:

- A new Apple Intelligence feature that will take a screenshot of the application and extract/summarize any text found.
- The new Genmoji feature, which runs out-of-process from the host application, where MAM data protections cannot be applied
- Some breaking changes that prevent the Intune iOS MAM SDK from blocking the Writing Tools feature in certain scenarios.

**Comment by ryanbland** [comment]
We've noticed with the new sdk users can no longer share their screens for things like zoom or take a screenshot for debugging, is there an app protection setting to allow this for certain workflows or is it now just a blanket no sharing your screen because of apple intelligence?  what is the expected workflow?

**Comment by Kyle-Reis** [comment]
@ryanbland - unfortunately on the client side there is currently no way for the SDK to differentiate between the different scenarios, so admins need to make the decision on whether to allow or prevent screen capture in general for each app. We are investigating the possibility of exposing an API to applications that would allow the application to acquire an image of its UI to enable scenarios where the data is kept within the application/user context. This wouldn't help with the screen sharing scenario you described though since it wouldn't apply to live sharing or to capturing/sharing the UI of other applications that may be protected with Intune app protection policies.

**Reply by ryanbland** [reply]
so If im using the zoom for intune i would still need to have the setting disabled to share the screen, or would apps that have the intune sdk built in work without having to turn on and off com.microsoft.intune.mam.screencapturecontrol?

**Reply by Kyle-Reis** [reply]
@ryanbland - Any Intune-enabled app that you would want shown in your screen sharing would need to have the setting disabled.

**Comment by aikensam** [comment]
Could you please add below reminder in SDK release notes of v19.7.6 and v20.2.1? A lot of customers are not aware the side effect of new screen capture control in the SDK.

https://learn.microsoft.com/en-us/mem/intune/apps/app-protection-policy-settings-ios#data-protection
![image](https://github.com/user-attachments/assets/6b9ee986-da6f-4514-931c-29df3ca23648)

**Reply by Kyle-Reis** [reply]
Hi @aikensam - In addition to the documentation you referenced above, information on this new functionality and how admins can control it have been communicated to end customers via the post and "What's New" documentation updates linked below:

https://iridias.microsoft.com/messagecenter?id=947829

https://learn.microsoft.com/mem/intune/fundamentals/whats-new#plan-for-change-blocking-screen-capture-in-the-latest-intune-app-sdk-for-ios-and-intune-app-wrapping-tool-for-ios

**Reply by ryanbland** [reply]
@Kyle-Reis does this "com.microsoft.intune.mam.screencapturecontrol" setting only apply to mam protected apps? we have users with intune mdm and i have added this setting to our config but we are still only seeing a white screen when in MDM. 
<img width="900" alt="Screenshot 2024-12-06 at 4 14 08 PM" src="https://github.com/user-attachments/assets/f9144005-53ba-4177-8e20-645dcc6e8265">

**Reply by Kyle-Reis** [reply]
Hi @ryanbland - The "com.microsoft.intune.mam.screencapturecontrol" app config setting can only be used to configure MAM based screen capture blocking and does not disable MDM screen capture controls. It must also be configured within MAM app configuration policy ("Managed apps" should be selected when creating the app config policy and NOT "Managed devices").

**Comment by aikensam** [comment]
The challenge is developers and Intune admins are two different teams. Those articles in learn.microsoft.com are not aware by developers when they update apps with our new releases of SDK. It will be appreciated if you can also highlight the important part in SDK release notes.

________________________________

**Comment by Yared-G** [comment]
> iOS 18.2 (currently in Beta) contains some new features and breaking changes that could allow an organization's data to leave the boundary of the Intune managed identity:
> 
> * A new Apple Intelligence feature that will take a screenshot of the application and extract/summarize any text found.
> * The new Genmoji feature, which runs out-of-process from the host application, where MAM data protections cannot be applied
> * Some breaking changes that prevent the Intune iOS MAM SDK from blocking the Writing Tools feature in certain scenarios.
> 
> The latest [20.2.1 release](https://github.com/microsoftconnect/ms-intune-app-sdk-ios/releases/tag/20.2.1) for Xcode 16 and [19.7.6 release](https://github.com/microsoftconnect/ms-intune-app-sdk-ios/releases/tag/19.7.6) for Xcode 15 addresses each of these scenarios with support for screen capture blocking, Genmoji blocking, and fixes for the Writing Tools issues.
> 
> We recommend that all application developers upgrade to the latest available supported release which corresponds to the Xcode version your application is currently compiled with, and that you push your updated application to production as soon as possible to ensure the fix for this issue is available before iOS 18.2 becomes generally available. If you have any questions, please [file an issue](https://github.com/microsoftconnect/ms-intune-app-sdk-ios/issues) in this repository or reply directly to this announcement.

@Kyle-Reis if we want to exempt an app that is used for screen sharing, will it allow us to share screen? we exempted one app, but it's not allowing us to see the screen. Thanks.

**Reply by Kyle-Reis** [reply]
Hi @Yared-G, exempting just the screen sharing app won't work. Unfortunately, there is no way to know the source of the screen capture, so any Intune-enabled app that you would want shown in your screen sharing app would need to have the MAM screen capture blocking disabled.
